src/main/java/com/adyen/model/checkout/PaymentsRequest.java

```diff
@@ -639,26 +615,26 @@ public PaymentsRequest addEncryptedCardData(String encryptedCardNumber, String e
      * @return paymentMethod
      */
     public PaymentsRequest addCardData(String cardNumber, String expiryMonth, String expiryYear, String securityCode, String holderName) {
-        this.paymentMethod = new HashMap<>();
-        this.paymentMethod.put(METHOD_TYPE, TYPE_SCHEME);
-        this.paymentMethod.put(NUMBER, cardNumber);
-        this.paymentMethod.put(EXPIRY_MONTH, expiryMonth);
-        this.paymentMethod.put(EXPIRY_YEAR, expiryYear);
+        DefaultPaymentMethodDetails paymentMethodDetails = new DefaultPaymentMethodDetails();
+        this.paymentMethod = paymentMethodDetails;
+
+        paymentMethodDetails.type(TYPE_SCHEME).number(cardNumber).expiryMonth(expiryMonth).expiryYear(expiryYear);
         if (securityCode != null) {
-            this.paymentMethod.put(CVC, securityCode);
+            paymentMethodDetails.setCvc(securityCode);
         }
         if (holderName != null) {
-            this.paymentMethod.put(HOLDER_NAME, holderName);
+            paymentMethodDetails.setHolderName(holderName);
         }
 
         return this;
     }
 
     public PaymentsRequest addOneClickData(String recurringDetailReference, String encryptedSecurityCode) {
-        this.paymentMethod = new HashMap<>();
-        this.paymentMethod.put(METHOD_TYPE, TYPE_SCHEME);
-        this.paymentMethod.put(RECURRING_DETAIL_REFERENCE, recurringDetailReference);
-        this.paymentMethod.put(ENCRYPTED_SECURITY_CODE, encryptedSecurityCode);
+        DefaultPaymentMethodDetails paymentMethodDetails = new DefaultPaymentMethodDetails();
+        this.paymentMethod = paymentMethodDetails;
```

Comment:
minor: exchange of 634 and 637 will make it more readable.

Reply:
done